src/main/java/org/takes/rs/RsPrettyXML.java

```diff
@@ -98,26 +111,94 @@ private Response make() throws IOException {
     }
 
     /**
-     * Format body with proper indents.
+     * Format body with proper indents using SAX.
      * @param body Response body
      * @return New properly formatted body
      * @throws IOException If fails
      */
     private static byte[] transform(final InputStream body) throws IOException {
-        final StreamSource source = new StreamSource(body);
+        final SAXSource source = new SAXSource(new InputSource(body));
         final ByteArrayOutputStream result = new ByteArrayOutputStream();
         try {
+            final XMLReader xmlreader = SAXParserFactory.newInstance()
+                .newSAXParser().getXMLReader();
+            source.setXMLReader(xmlreader);
+            xmlreader.setFeature(
+                RsPrettyXML.LOAD_EXTERNAL_DTD, false
+            );
             final Transformer transformer = TransformerFactory.newInstance()
                 .newTransformer();
             // @checkstyle MultipleStringLiteralsCheck (2 line)
             transformer.setOutputProperty(
                 OutputKeys.OMIT_XML_DECLARATION, "yes"
             );
+            RsPrettyXML.prepareDocType(body, transformer);
             transformer.setOutputProperty(OutputKeys.INDENT, "yes");
             transformer.transform(source, new StreamResult(result));
         } catch (final TransformerException ex) {
             throw new IOException(ex);
+        } catch (final SAXException ex) {
+            throw new IOException(ex);
+        } catch (final ParserConfigurationException ex) {
+            throw new IOException(ex);
         }
         return result.toByteArray();
     }
+
+    /**
+     * Parses body to get DOCTYPE and configure Transformer
+     * with proper method, public id and system id.
+     * @param body The body to be parsed.
+     * @param transformer Transformer to configure with proper properties.
+     * @throws IOException if something goes wrong.
+     */
+    private static void prepareDocType(final InputStream body,
+        final Transformer transformer) throws IOException {
+        try {
+            final DocumentType doctype = RsPrettyXML.getDocType(body);
+            if (null != doctype) {
+                if (null == doctype.getSystemId()
+                    && null == doctype.getPublicId()
+                    // @checkstyle MultipleStringLiteralsCheck (3 line)
+                    && "html".equalsIgnoreCase(doctype.getName())) {
+                    transformer.setOutputProperty(OutputKeys.METHOD, "html");
+                    transformer.setOutputProperty(OutputKeys.VERSION, "5.0");
+                } else {
```

Comment:
html5 specification says that it can have only <!DOCTYPE html> or <!DOCTYPE HTML>. So if at least one id is not null - it will be not html5 and will be parsed as html4 or xhtml.